src/main/java/fi/aalto/cs/apluscourses/ui/repl/ReplConfigurationForm.java

```diff
@@ -0,0 +1,128 @@
+package fi.aalto.cs.apluscourses.ui.repl;
+
+import static fi.aalto.cs.apluscourses.presentation.ReplConfigurationFormModel.showREPLConfigWindow;
+import static java.util.Objects.requireNonNull;
+
+import com.intellij.openapi.fileChooser.FileChooserDescriptor;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.ui.ComboBox;
+import com.intellij.openapi.ui.TextFieldWithBrowseButton;
+import com.intellij.openapi.vfs.VirtualFile;
+import fi.aalto.cs.apluscourses.presentation.ReplConfigurationFormModel;
+import javax.swing.JCheckBox;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+
+public class ReplConfigurationForm extends JPanel {
+
+  public static final String INFOLABEL_TEXT = "<html>"
+      + "This is an A+ Scala REPL configuration window. By default, the working "
+      + "directory and the dependencies, loaded to the REPL classpath belong to the Module, that "
+      + "it was started on. To change the behavior, use the checkbox at the bottom of the window."
+      + "</html>";
+
+  private ReplConfigurationFormModel model;
```

Comment:
@OlliKiljunen can you take a look here and at the overall composition. I'm using a model for data inside the UI (kinda Presentation), but is this a correct way?

Reply:
To me this seems very good.

I'd argue that there is no Model-View-Presenter/ViewModel pattern here, but that is by no means a problem because this view does not actually handle our domain model classes. Instead, the pattern used here is sometimes called an "Autonomous View" and as far as I understand, this implementation of yours could be a textbook example of it done right. :)

Reply:
thanks...
so this sh*t (implementation) even has a name :rofl: , I'll look it up